beastling/cli.py

```diff
@@ -79,76 +84,68 @@ def main(*args):
-    sys.exit(0)
+    exit(status=0)
 
 
 def do_extract(args):
     if len(args.config) != 1:
-        errmsg("Can only extract from exactly one BEAST XML file")
-        sys.exit(1)
-    if not os.path.exists(args.config[0]):
-        errmsg("No such BEAST XML file: %s\n" % args.config)
-        sys.exit(2)
+        exit(msg="Can only extract from exactly one BEAST XML file", status=1)
+    if not args.config[0].exists():
+        exit(msg="No such BEAST XML file: %s" % args.config, status=2)
     try:
-        messages = extract(args.config[0], args.overwrite)
+        for msg in extract(args.config[0], args.overwrite):
+            sys.stdout.write(msg)
     except wrap_errors as e:
-        errmsg("Error encountered while extracting BEASTling config and/or data files:\n")
-        traceback.print_exc()
-        sys.exit(3)
-    for msg in messages:
-        sys.stdout.write(msg)
+        exit(
+            msg="Error encountered while extracting BEASTling config and/or data files:",
+            status=3,
+            exception=True)
 
 
 def do_generate(args):
 
     # Make sure the requested configuration file exists
     for conf in args.config:
-        if not os.path.exists(conf):
-            errmsg("No such configuration file: %s\n" % conf)
-            sys.exit(1)
+        if not conf.exists():
+            exit(msg="No such configuration file: %s" % conf, status=1)
 
     # Build but DON'T PROCESS the Config object
     # This is fast, and gives us enough information to check whether or not
     try:
         config = Configuration(
             configfile=args.config, stdin_data=args.stdin, prior=args.prior, force_glottolog_load=args.report)
     except wrap_errors as e: # PRAGMA: NO COVER
-        errmsg("Error encountered while parsing configuration file:\n")
-        traceback.print_exc()
-        sys.exit(2)
+        exit(msg="Error encountered while parsing configuration file:", status=2, exception=True)
 
     # Make sure we can write to the appropriate output filename
     output_filename = pathlib.Path(args.output) if args.output else config.admin.path(".xml")
     if output_filename.exists() and not args.overwrite:
-        errmsg("File %s already exists!  Run beastling with the --overwrite option if you wish to overwrite it.\n" % output_filename)
-        sys.exit(4)
+        exit(msg="File %s already exists! Run beastling with the --overwrite option if you wish "
+                 "to overwrite it." % output_filename,
+             status=4)
 
     # Now that we know we will be able to save the resulting XML, we can take
     # the time to process the config object
     try:
         config.process()
-    except wrap_errors as e:
-        errmsg("Error encountered while parsing configuration file:\n")
-        traceback.print_exc()
-        sys.exit(2)
+    except wrap_errors as e:  # pragma: no cover
+        exit(msg="Error encountered while parsing configuration file:", status=2, exception=True)
```

Comment:
I really like the new `exit`, but this leaves me wondering whether there should be/how there could be a central instance that matches status codes to exit messages.

Reply:
So basically a man page?

Reply:
I was thinking on the Python side – so an Enum or similar. (That could of course also be used to generate a man page or an argumentparser.Description section, but that is not directly what I meant.)

Reply:
Yes, I understand. I wanted to highlight that this is a documentation issue, and why we might want to handle this more consistently. So I agree. I don't use `Enum`s much, but considering the support they have in IDEs they might make sense here - but a plain `dict` mapping return code to error message might do as well.